File: bank-service/src/main/java/rs/edu/raf/dto/CurrentBankAccountCreateDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;

import java.math.BigDecimal;

public record CurrentBankAccountCreateDTO (
        @NotNull String JBMG,
        @NotNull String accountType,
        @PositiveOrZero BigDecimal maintenancePrice
) { }

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/CurrentBankAccountCreateDTO.kt
package org.example.bankservicekotlin.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.PositiveOrZero
import java.math.BigDecimal

data class CurrentBankAccountCreateDTO(
    @NotNull val JMBG: String,
    @NotNull val accountType: String,
    @PositiveOrZero val maintenancePrice: BigDecimal
)
